### PR TITLE
Fix cluster-wide MachineType

### PR DESCRIPTION
**Pull request by acardace**
**What this PR does / why we need it**:
Since https://github.com/kubevirt/kubevirt/pull/8293 there is a new field for setting the cluster-wide MachineType, let's use that along the old one (for compatibility reasons).

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
https://issues.redhat.com/browse/CNV-30288
```

**Release note**:

```release-note
Fix cluster-wide machineType
```

### CODE REVIEW

**Comment by nunnatsa** [comment]
/test hco-e2e-kv-smoke-gcp
/test hco-e2e-operator-sdk-gcp
/test okd-hco-e2e-operator-sdk-gcp
/ test okd-hco-e2e-upgrade-operator-sdk-gcp

File: controllers/operands/kubevirt.go [hunk 1]
```diff
@@ -426,6 +431,14 @@ func getKVConfig(hc *hcov1beta1.HyperConverged) (*kubevirtcorev1.KubeVirtConfigu
 	if val, ok := os.LookupEnv(machineTypeEnvName); ok {
 		if val = strings.TrimSpace(val); val != "" {
 			config.MachineType = val
+
+			config.ArchitectureConfiguration = &kubevirtcorev1.ArchConfiguration{
+				Amd64: &kubevirtcorev1.ArchSpecificConfiguration{
+					MachineType:      val,
+					OVMFPath:         DefaultAMD64OVMFPath,
+					EmulatedMachines: strings.Split(DefaultAMD64EmulatedMachines, ","),
```
Comment (tiraboschi) [inline controllers/operands/kubevirt.go:439]: Currently `MachineType` takes the value of  `MACHINETYPE` env variable that can be set at bundle build time.
Are `OVMFPath` and `EmulatedMachines` completely independent from `MachineType` or we should also introduce 2 additional optional env variables for them?

Comment (acardace) [reply controllers/operands/kubevirt.go:439]: They are 3 independent settings, we can, of course, give the option to set all 3 with env vars but I thought that for the sake of simplicity to just hardcode the remaining 2 to KubeVirt's default.

**Comment by acardace** [comment]
@tiraboschi any idea why pull-hyperconverged-cluster-operator-e2e-k8s-1.27  keeps failing?

**Comment by tiraboschi** [comment]
> @tiraboschi any idea why pull-hyperconverged-cluster-operator-e2e-k8s-1.27 keeps failing?

it failed on:
```
+ /home/prow/go/src/github.com/kubevirt/hyperconverged-cluster-operator/_kubevirtci/cluster-up/up.sh
selecting podman as container runtime
Error response from daemon: normalizing image: normalizing name for compat API: invalid reference format
make: *** [Makefile:128: cluster-up] Error 1
```

not really sure about the root cause.
Let's retry to see if it got fixed on kubevirtci side.

**Review by tiraboschi — APPROVED** [review]

**Pull request merged**